File: LeetCode_Algorithm/1275(나중에 한 번 더 보기)/1275.js
var tictactoe = function(moves) {
  // sol1 O(mn) O(n^2) 70~93ms 76~36%
  Board = new Array(3);
  for (let i=0; i < 3; i++){
    Board[i] = new Array(3)
  }
  for (let i=0; i < moves.length; i++) {
    i % 2 == 0 ? val = "A" : val = "B"
    let r = moves[i][0]
    let c = moves[i][1]
    Board[r][c] = val
  }
  for (let r of [0,1,2]){
    for (let c of [0,1,2]){
      ans = findWinner(r,c)
      if (ans != undefined){
        return ans
      }
    }
  }

  return moves.length == 9 ? "Draw" : "Pending"

};
function findWinner(r,c){
  chr = Board[r][c]
  dr = [0,1,0,-1, -1,1,1,-1]
  dc = [1,0,-1,0, 1,1,-1,-1]


  for(let i=0; i < 8; i++){
    var r2 = r+dr[i]; var c2 = c+dc[i]; var cnt = 1;
    while (0 <= c2 && c2 < 3 && 0 <= r2 && r2 < 3){
      if (Board[r2][c2] == undefined || Board[r2][c2] != chr){
        break
      }else{
        cnt += 1
        if (cnt == 3){
          return Board[r2][c2]
        }
      }
      r2 += dr[i]; c2 += dc[i]
    }
  }
}
console.log(tictactoe([[0,0],[1,1],[2,0],[1,0],[1,2],[2,1],[0,1],[0,2],[2,2]]))

File: LeetCode_Algorithm/1275(나중에 한 번 더 보기)/1275.py
from typing import List

class Solution:
    def tictactoe(self, moves: List[List[int]]) -> str:
        # sol1 O(mn) O(n^2) 45~52ms 57~38%
        # m은 input의 길이 n은 보드의 가로 길이
        # board에 다 넣고 dfs로 찾기
#         global board
#         board =  [[None for _ in range(3)] for _ in range(3)]

#         for i in range(len(moves)):
#             chr = "A" if i % 2==0 else "B"
#             r,c = moves[i]
#             board[r][c] = chr

#         for r in range(3):
#             for c in range(3):
#                 if board[r][c] != None:
#                     ans = findWinner(r,c)
#                     if ans is not None:
#                         return ans
#         return "Draw" if len(moves) == 9 else "Pending"


# def findWinner(r,c):

#     chr = board[r][c]
#     dr = [0,1,0,-1,-1,1,1,-1]
#     dc = [1,0,-1,0,1,1,-1,-1]
#     cnt = 1

#     for i in range(8):
#         r2, c2, cnt2 = r +dr[i], c + dc[i], cnt
#         while  0 <= r2 < 3 and 0 <= c2 < 3:
#             if board[r2][c2] == chr:
#                 cnt2 += 1
#                 if cnt2 == 3:
#                     return board[r2][c2]
#                 r2 += dr[i]; c2 += dc[i]
#             else:
#                 break

        pass



        


    
        


    
 

print(Solution().tictactoe([[0,0],[1,1]]))